core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/EntityApiServiceSpec.scala

```diff
@@ -741,6 +741,79 @@ class EntityApiServiceSpec extends ApiServiceSpec {
     assertResult(activeAttributeCount1)(activeAttributeCount2)
   }
 
+  it should "return 204 when deleting an entity type that has no references to it" in withTestDataApiServices { services =>
+
+    val entities = Seq.tabulate(100){ i =>
+      Entity(s"foo$i", "typeToDelete", Map(AttributeName.withDefaultNS("hello") -> AttributeString("world")))
+    }
+
+    runAndWait(entityQuery.save(testData.workspace, entities))
+
+    val (entityCountBefore, attributeCountBefore) = countEntitiesAttrs(testData.workspace)
+    val (activeEntityCountBefore, activeAttributeCountBefore) = countActiveEntitiesAttrs(testData.workspace)
+
+    Delete(s"${testData.workspace.path}/entityTypes/typeToDelete") ~>
+      sealRoute(services.entityRoutes) ~>
+      check {
+        assertResult(StatusCodes.NoContent) {
+          status
+        }
+        assertResult(Seq.empty) {
+          runAndWait(entityQuery.listActiveEntitiesOfType(testData.workspace, "typeToDelete")).toSeq
+        }
+      }
+
+    val (entityCountAfter, attributeCountAfter) = countEntitiesAttrs(testData.workspace)
+    val (activeEntityCountAfter, activeAttributeCountAfter) = countActiveEntitiesAttrs(testData.workspace)
+
+    //Ensure that the entity and attribute count (including "deleted" entities is the same before and after
+    assertResult(entityCountBefore)(entityCountAfter)
+    assertResult(attributeCountBefore)(attributeCountAfter)
+
+    //Ensure that we have marked the appropriate number of entities and attributes as "deleted"
+    assertResult(activeEntityCountBefore - entities.size)(activeEntityCountAfter)
+    assertResult(activeAttributeCountBefore - entities.flatMap(_.attributes).size)(activeAttributeCountAfter)
```

Comment:
I could be misunderstanding, but it looks like this only works because each entity has one attribute. Should this sum up the number of attributes for each entity?

Reply:
Another way to write this would be `entities.map(_.attributes).flatten.size`

`entities.map(_.attributes)` would return a `Seq[Map[AttributeName, Attribute]]`, which would be a list of all of the attribute lists for all of the `entities`. Calling `.flatten` on that then makes it a `Seq[(AttributeName, Attribute)]`, which is the list of all attributes across all of the `entities`, which we can then take the size of to get the total count of attributes. 

I used `.flatMap` which combines the `.map` and subsequent `.flatten` into one step